lib/Cake/Test/Case/Core/ConfigureTest.php

```diff
@@ -27,12 +27,14 @@
  */
 class ConfigureTest extends CakeTestCase {
 
+	protected $_restoreError = false;
```

Comment:
What do we need it for? :thinking:

Reply:
Short answer.
"A mechanism to return to PHPUnit's error handler from the error handler set in `Configure::bootstrap()`".
The code was copied from `ErrorHandlerTest`.

--

In [`bootstrap.php`](https://github.com/tenkoma/cakephp2-php8/blob/test/lib/Cake/Test/bootstrap.php) for testing, set the PHPUnit error handler as follows.

```php
set_error_handler(new \PHPUnit\Util\ErrorHandler(true, true, true, true));
```

PHPUnit's error handler will not print to the screen when a PHP error occurs, but will instead throw an exception such as `PHPUnit\Framework\Error\Notice` or `PHPUnit\Framework\Error\Warning`.

By catching the thrown exceptions outside of the test method, assertions such as `expectWarning()` can be made.

However, in the following test log, the PHP Warning is output as is.
https://github.com/tenkoma/cakephp2-php8/actions/runs/3862552514/jobs/6584192189

```
(snip)
........W................W...W.WW............................ 1525 / 4070 ( 37%)
................FWarning: 512 :: Column type varchar(255) does not exist on line 3505 of CORE/Cake/Model/Datasource/DboSource.php
Trace:
DboSource::buildColumn() - CORE/Cake/Model/Datasource/DboSource.php, line 3505
MysqlTest::testBuildColumnBadType() - CORE/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php, line 3246
PHPUnit\Framework\TestCase::runTest() - ROOT/vendors/phpunit/phpunit/src/Framework/TestCase.php, line 1548
(snip)
```

This Warning error occurred during the execution of `MysqlTest::testBuildColumnBadType()`.

I thought that the Warning error was the result of the error handler being overridden before this error occurred.
So I searched for the cause and restored the error handler in `ConfigureTest`, and the test log is now output correctly.

Reply:
Got it, thanks for explanation! Resolving.